src/items/generics.md

```diff
@@ -83,6 +83,98 @@ where
 }
 ```
 
+## Formulae and Free Variables
+
+The syntatic locations a *variable* is allowed is a *formulae*. Formulae for
+type and lifetime variables are:
+
+* [Types]
+* Type constructors (for example, `Option` in `enum Option<T> { ... }`)
+* Lifetimes (only lifetime variables)
+* [Items]
+* Item constructors (for example, `Into` in `trait Into<T>`)
+* [Associated Items]
+* [Trait Bounds]
+* [Expressions] \(through path expressions with generics)
+
+The free variables of a formulae are those not referenced (*bound*) by a generic
+binder in the formulae.
+
+These free variables may be substituted for other types or lifetimes.
+A formulae is *concrete*, or *closed*, if there are no free variables in it.
+See [Wikipedia][wikipedia free variables] for more.
+
+Examples of generic binders for type and lifetime variables are:
+
+* `for<'a>` in `for<'a> fn(&'a u8)`
+* `<T>` in `fn identity<T>(x: T) -> T { x }` for item constructors
+* [Traits] and [implementations] implicitly bind `Self`
+
+All type and lifetimes are unbound at the boundaries of items.
```

Comment:
Also not sure what this sentence is saying...

Reply:
T'was the last thing I wrote before my brain turned off. I need to expand it, but it's the fact that in the following example, `CONST` has no free variables, whether implicitly `Self` from traits or the generic parameter `T` of the function itself:

```rust
fn generic<T>(t: T) {
    const CONST: () = ();
}
```

Reply:
Ah yes; that's a good point, we don't have scoped type variables. Be careful not to imply that this applies to `bar` in `trait Foo { fn bar(); }` because `Foo` and `bar` are both in the syntactic category of items here.